src/library/tabledelegates/keydelegate.cpp

```diff
@@ -0,0 +1,61 @@
+#include "library/tabledelegates/keydelegate.h"
+
+#include <qnamespace.h>
+
+#include <QPainter>
+#include <QStyle>
+#include <QTableView>
+
+#include "moc_keydelegate.cpp"
+#include "track/keyutils.h"
+#include "util/color/predefinedcolorpalettes.h"
+#include "util/color/rgbcolor.h"
+
+KeyDelegate::KeyDelegate(QTableView* pTableView)
+        : TableItemDelegate(pTableView) {
+}
+
+void KeyDelegate::paintItem(
+        QPainter* painter,
+        const QStyleOptionViewItem& option,
+        const QModelIndex& index) const {
+    paintItemBackground(painter, option, index);
+
+    QString keyText = index.data().toString();
+    mixxx::track::io::key::ChromaticKey key = KeyUtils::guessKeyFromText(keyText);
+    int openKeyNumber = KeyUtils::keyToOpenKeyNumber(key);
+
+    if (openKeyNumber != 0) {
+        const auto rgbColor =
+                mixxx::PredefinedColorPalettes::kDefaultKeyColorPalette.at(
+                        openKeyNumber - 1); // Open Key numbers start from 1
```

Comment:
check bounds as well.
```suggestion
        const auto& palette = mixxx::PredefinedColorPalettes::kDefaultKeyColorPalette;
        DEBUG_ASSERT(openKeyNumber <= palette.length());
        const auto rgbColor =
                palette.at(
                        openKeyNumber - 1); // Open Key numbers start from 1
```